trustfall_core/src/graphql_query/query.rs

```diff
@@ -125,49 +125,59 @@ fn try_get_query_root(document: &ExecutableDocument) -> Result<&Positioned<Field
 
     match &document.operations {
         DocumentOperations::Multiple(mult) => {
-            return Err(ParseError::MultipleOperationsInDocument(mult.values().next().unwrap().pos))
+            return if mult.values().len() > 1 {
+                mult.values().next().expect("Could not iterate to first value in document.");
```

Comment:
I'm confused by this line, since it seems to be a no-op. Did you mean to assign its result and use it in the `Err()` below?

Reply:
Also, the `return if` is not very idiomatic, and I found it surprising -- I even missed it the first time because I didn't expect it there. Could we avoid it by putting the `return` inside each block of the `if`?

Reply:
I guess the methodology behind this is that its meant to mutate the iterator. The first call to `.next()` will consume the first value (i.e. the first document) the next call to `.next()` would be the second value (the value of the operation that would trigger the `MultipleOperationsInDocument` error.)

Reply:
This may not be a good way to do this though and am happy to refactor :)

Reply:
@obi1kenobi weird the page didn't load your comment about the `return if` at first 😆 Will take a look - this was actually a suggestion from `clippy` as a linting thing. I wonder if it may be worth it to add a linting rule/exception from clippy to avoid recommending `return if` then?

Reply:
Ah I completely missed that, clearly.

Usually, discarding a return value is done by `let _ = call_whose_value_we_discard()`, with a comment above it explaining what's going on. This will silence any clippy `#[must_use]` errors. In this case, I think the `.expect()` prevented that clippy lint, but I don't think it should have -- so better safe than sorry + follow convention.

Reply:
`nth()` is fine too!

Reply:
`return if` is tricky. If the blocks are tiny, it can be very good! In this case, clippy was probably stuck in a local maximum -- but perhaps there's a better way to do it overall.

Reply:
> `return if` is tricky. If the blocks are tiny, it can be very good! In this case, clippy was probably stuck in a local maximum -- but perhaps there's a better way to do it overall.

This is really good to know.